src/ngraph/op/quantized_convolution.cpp

```diff
@@ -0,0 +1,196 @@
+        INPUT_SCALE,
+        INPUT_ZERO_POINT,
+        FILTER_SCALE,
+        FILTER_ZERO_POINT,
+        OUTPUT_SCALE,
+        OUTPUT_ZERO_POINT
+    };
+
+    NODE_VALIDATION_CHECK(
+        this, m_output_type.is_static(), "Output element type must not be dynamic");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_output_type.is_quantized(),
+                          "Output element type (",
+                          m_output_type,
+                          ") must be a quantized type");
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(INPUT).is_quantized(),
+                          "Input element type (",
+                          get_input_element_type(INPUT),
+                          ") must be a quantized type");
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(FILTER).is_quantized(),
+                          "Filter element type (",
+                          get_input_element_type(FILTER),
+                          ") must be a quantized type");
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(INPUT_SCALE).is_real() ||
+                              get_input_element_type(INPUT_SCALE).is_dynamic() ||
+                              get_input_element_type(FILTER_SCALE).is_real() ||
+                              get_input_element_type(FILTER_SCALE).is_dynamic() ||
+                              get_input_element_type(OUTPUT_SCALE).is_real() ||
+                              get_input_element_type(OUTPUT_SCALE).is_dynamic(),
+                          "Scale must be a floating point number");
+
+    NODE_VALIDATION_CHECK(
+        this,
+        get_input_element_type(0).compatible(get_input_element_type(INPUT_ZERO_POINT)),
+        "Input Zero point element type (",
+        get_input_element_type(INPUT_ZERO_POINT),
+        ") must match input element type (",
+        get_input_element_type(0),
+        ")");
+
+    NODE_VALIDATION_CHECK(
+        this,
+        get_input_element_type(1).compatible(get_input_element_type(FILTER_ZERO_POINT)),
+        "Filter Zero point element type (",
+        get_input_element_type(FILTER_ZERO_POINT),
+        ") must match filter element type (",
+        get_input_element_type(1),
+        ")");
+
+    // TODO Remove these checks once we support channelwise and vector of scales
+    NODE_VALIDATION_CHECK(this,
+                          get_input_partial_shape(2).compatible(PartialShape{}) &&
+                              get_input_partial_shape(3).compatible(PartialShape{}),
+                          "Input scale and input zero point shape must be same and 1");
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_partial_shape(4).compatible(PartialShape{}) == 1 &&
+                              get_input_partial_shape(5).compatible(PartialShape{}) == 1,
+                          "Filter scale and filter zero point shape must be same and 1");
```

Comment:
Why "==1"?

Reply:
I think this must be residue from the old check, which was `shape_size(foo) == 1`?

Reply:
yes, it was :)